app/controllers/api/v1/packages_controller.rb

```diff
@@ -270,16 +270,36 @@ def remove_stockit_prefix(stockit_inventory_number)
       def package_params
         get_package_type_id_value
         set_favourite_image if @package && !@package.new_record?
-        attributes = [:quantity, :length, :width, :height, :weight, :pieces, :notes, :item_id,
-          :received_at, :rejected_at, :package_type_id, :state_event,
-          :inventory_number, :designation_name, :donor_condition_id, :grade,
-          :location_id, :box_id, :pallet_id, :stockit_id,
-          :order_id, :stockit_designated_on, :stockit_sent_on,
-          :case_number, :allow_web_publish, :received_quantity, :state,
-          packages_locations_attributes: [:id, :location_id, :quantity]]
+        attributes = [:quantity, :length, :width, :height, :weight, :pieces,
+          :notes, :item_id, :received_at, :rejected_at, :package_type_id,
+          :state_event, :inventory_number, :designation_name, :donor_condition_id,
+          :grade, :location_id, :box_id, :pallet_id, :stockit_id, :order_id,
+          :stockit_designated_on, :stockit_sent_on, :case_number, :allow_web_publish,
+          :received_quantity, :state, :detail_type, :detail_id,
+          packages_locations_attributes: [:id, :location_id, :quantity],
+          detail_attributes: [computer_attributes, electrical_attributes,
+          computer_accessory_attributes].flatten.uniq]
         params.require(:package).permit(attributes)
       end
 
+      def computer_attributes
+        [:brand, :model, :serial_num, :country_id, :size,
+          :cpu, :ram, :hdd, :optical, :video, :sound, :lan, :wireless,
+          :usb, :comp_voltage, :os, :os_serial_num, :comp_test_status, :ms_office_serial_num,
+          :mar_os_serial_num, :mar_ms_office_serial_num, :updated_by_id]
+      end
+
+      def electrical_attributes
+        [:brand, :model, :serial_number, :country_id, :standard,
+          :voltage, :frequency, :power, :system_or_region, :test_status,
+          :tested_on, :updated_by_id]
+      end
+
+      def computer_accessory_attributes
+        [:brand, :model, :serial_num, :country_id, :size,
+          :interface, :comp_voltage, :comp_test_status, :updated_by_id]
```

Comment:
**NBD**: It would be good if we can arrange these attributes in alphabetical order, so that it would be good to track in future

Reply:
sure @swatijadhav , i'll do it now.